pkg/keystone/statefulset.go

```diff
@@ -0,0 +1,108 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keystone
+
+import (
+	keystonev1beta1 "github.com/openstack-k8s-operators/keystone-operator/api/v1beta1"
+	common "github.com/openstack-k8s-operators/lib-common/pkg/common"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StatefulSet func
+func StatefulSet(
```

Comment:
looking at the volume mounts in below the fernet keys are mounted readonly and I don't see any other volumes that would require this to be a statefulset instead of a deployment so can you either document in a comment why this need to be a stateful set or revert this to a deployment?

Reply:
see above, just the naming for listing services. I'll switch it back if we do not want it.

Reply:
moved back to use a deployment